frontend/lib/resolution/InitResolver.cpp

```diff
@@ -279,13 +279,15 @@ static const Type* ctFromSubs(Context* context,
     auto oldBasic = cls->basicClassType();
     CHPL_ASSERT(oldBasic && "Not handled!");
 
-    auto instantatiatedFrom = subs.size() == 0 ? nullptr :
-                                                 root->toBasicClassType();
+    bool genericParent = superType && superType->substitutions().size() != 0;
```

Comment:
Now that you've pointed out this pattern, I wonder if it would break down if there was just a generic grandparent, where the parent was also concrete? Can you add a test for that case to see if this works there?

If not, we might need to adjust this to check ``superType->instantiatedFromCompositeType() != nullptr`` instead. In a way, this might be more conceptually in line with the overall spirit of this change.

Reply:
Indeed it does break down with a generic grandparent. Added a test and adjusted to handle this case.